catalog/service/rest/src/main/java/org/projectnessie/catalog/service/rest/IcebergApiV1GenericResource.java

```diff
@@ -138,14 +145,34 @@ public Uni<Void> commitTransaction(
     commitTransactionRequest.tableChanges().stream()
         .map(
             tableChange -> {
+              List<IcebergMetadataUpdate> updates =
+                  tableChange.updates().stream()
+                      .filter(update -> !(update instanceof SetLocation))
+                      .collect(Collectors.toList());
+
               ContentKey key = requireNonNull(tableChange.identifier()).toNessieContentKey();
 
-              return IcebergCatalogOperation.builder()
-                  .updates(tableChange.updates())
-                  .requirements(tableChange.requirements())
-                  .key(key)
-                  .type(ICEBERG_TABLE)
-                  .build();
+              IcebergCatalogOperation.Builder builder =
+                  IcebergCatalogOperation.builder()
+                      .updates(updates)
+                      .requirements(tableChange.requirements())
+                      .key(key)
+                      .type(ICEBERG_TABLE);
+
+              if (tableChange.hasAssertCreate()) {
+                List<IcebergUpdateRequirement> invalidRequirements =
+                    tableChange.requirements().stream()
+                        .filter(req -> !(req instanceof IcebergUpdateRequirement.AssertCreate))
+                        .collect(Collectors.toList());
+                checkArgument(
+                    invalidRequirements.isEmpty(),
+                    "Invalid create requirements: %s",
+                    invalidRequirements);
+                String location = defaultTableLocation(decoded.warehouse(), key);
+                builder.addUpdate(setLocation(location));
```

Comment:
What if `AssertCreate` is missing? Aren't we losing the `SetLocation` update in that case?

Reply:
Yes we are. That's the idea: ignore any locations sent by the client, and use our own.

Reply:
I mean should we not do `builder.addUpdate(setLocation(defaultLocation))` even if/when `AssertCreate` is not present?

Reply:
I think not, because in this case it's an update, so the table already exists at its default location. Am I missing something?

Reply:
To clarify: `AssertCreate` is a requirement that the client chooses to submit. Can we be sure that a table cannot be created without submitting an `AssertCreate`?

Reply:
While it seems that it is not possible to create a table without an `AssertCreate` now, it seems to me that it's a co-incidence rather than a deliberate check. From the REST API perspective an `AssertCreate` does not have to be present. 

Should we set the location in `CatalogServiceImpl` perhaps rather than at the REST level?

Reply:
Yes I was leaning towards that already, and you convinced me. I will revisit this PR entirely.